peer.go

```diff
@@ -63,6 +64,64 @@ func (id ID) MatchesPublicKey(pk ic.PubKey) bool {
 	return oid == id
 }
 
+func (id ID) ExtractEd25519PublicKey() ic.PubKey {
+	// ed25519 pubkey identity format
+	// <identity mc><length (2 + 32 = 34)><ed25519-pub mc><ed25519 pubkey>
+	// <0x00       ><0x22                ><0xed01        ><ed25519 pubkey>
```

Comment:
Is there a Go function that will varint encode a byte slice for me?

The ed25519-pub code `0xed` is `0b11101101` in binary. The last byte needs to have it's MSB set to 0 (according to [this](https://richardschneider.github.io/net-ipfs-core/api/Ipfs.Varint.html)) and each byte except the last contains 7 bits of information. So I imagine the MSB (which is 1) gets chopped off from the first byte and added to the next byte. That would yield `0b(1)1101101(0)1000000`. Or is it that we shift everything by one bit, yielding `0b(1)1110110(0)1000000`? I'm so confused by varint!